lib/kcerts/ssh.go

```diff
@@ -219,7 +315,21 @@ func FindSSHAgentFromEnv() *SSHAgent {
 // CreateNewSSHAgent creates a new ssh agent. Its env variables have not been added to the shell. It does not maintain
 // its own connection.
 func CreateNewSSHAgent() (*SSHAgent, error) {
-	cmd := exec.Command("ssh-agent", "-s")
+	a := &SSHAgent{}
+
+	socket, err := a.GetStandardSocketPath()
+	if err != nil {
+		return nil, err
+	}
+
+	os.Remove(socket) // ignore errors
+	_, err = os.Stat(socket)
+	if err == nil {
+		return nil, fmt.Errorf("unable to delete existing socket: %s", socket)
+	}
+
+	// TODO(jonathan): is it necessary or safer to delete the socket first?
+	cmd := exec.Command("ssh-agent", "-s", "-a", socket)
```

Comment:
now that we use a symlink, why not invoke `ssh-agent` as usual? I'm worried that otherwise we have two different "landing" ssh-agent configs. In some cases, our socket will be a symlink, in some cases, it will be directly the socket.

But... one configuration will work well in some cases, and not in others, and vice-versa, we'll pretty much end up having to debug and ensure that all programs work in all cases "oh, he's only seeing this problem when it's a symlink - that code is hardcoding the agent path to be in /tmp, or ... it doesn't allow for symlink or ...".

Reply:
(and there's no race conditions around socket deletion/removal and agent starting)

Reply:
Done.